### DISCUSSION TITLE
[DONE] Getting list of COGs for a given quadkey

### DISCUSSION THREAD

**Discussion by mackdelany** [developmentseed/titiler#350] [Feature Request]
Howdy,

As far as I'm aware, there is no way to ask Titiler to query a MosaicJSON file and return a list of COGs that overlap a given quadkey. This would be helpful for our use case at [CarbonCrop](https://carboncrop.nz).

If this is correct (I may have missed something) and the community is happy, I will submit a PR to expose this functionality.

My initial inclination would probably be something like:
- Add a `mosaicjson/cogs` endpoint
- Have the endpoint require a `quadkey` query param
- Have the endpoint return all valid COG paths for the given quadkey

So a request like:

`http://<titiler-host>/mosaicjson/cogs/?url=<path-to-mosaic-file>/mosaic.json&quadkey=311330301132`

Might return something like:

```
[
      "s3://super-awesome-cog-1.tif",
      "s3://super-awesome-cog-2.tif",
      "s3://super-awesome-cog-3.tif"
]
```

Definitely keen to hear if there are other preferences / or reasons why this isn't a thing / this does exist and I've missed it.

Thanks

**Comment by mackdelany** [comment]
After thinking about this a little more I think it's probably better to use a bounding box or lat long, something like:

`http://<titiler-host>/mosaicjson/cogs/{minx},{miny},{maxx},{maxy}`

`http://<titiler-host>/mosaicjson/cogs/{lon},{lat}`

With Titiler returniing lists of COGs which encompass the bbox or lat long.

**Reply by vincentsarago** [reply]
We could add all those 3 endpoints 

```
http://<titiler-host>/mosaicjson/{minx},{miny},{maxx},{maxy}/assets

http://<titiler-host>/mosaicjson/{lon},{lat}/assets

http://<titiler-host>/mosaicjson/{quadkey}/assets
```

**Reply by geospatial-jeff** [reply]
These would be really useful, would most of this code live in cogeo-mosaic?

**Reply by vincentsarago** [reply]
Yes you'll need to use `{Backend}.asset_for_tile` and `{Backend}.asset_for_point` methods 
https://github.com/developmentseed/cogeo-mosaic/blob/master/cogeo_mosaic/backends/base.py#L119-L126

Note: for `bbox` search you'll need to add a bit more code to get a list of quadkey for the given bbox for the mosaic `quadkey_zoom`.

**Reply by vincentsarago** [reply]
Also Side note: we can get the list of assets used per tile/point in the response headers https://github.com/developmentseed/titiler/blob/master/src/titiler/mosaic/titiler/mosaic/factory.py#L250-L251

**Reply by mackdelany** [reply]
Cool, this would be great. I'll send a PR across tomorrow. Thanks!

**Comment by vincentsarago** [comment]
work done in #351 by @mackdelany 🙏

**Comment by vincentsarago** [comment]
@mackdelany I merged your PR but now I'm wondering if using `/{quadkey}/assets` is a good approach. What do you think about using `/{z}/{x}/{y}/assets` ?

**Reply by mackdelany** [reply]
Yeah I think that's a better approach. Thanks for updating so quickly!
